packages/astro/src/runtime/client/dev-overlay/plugins/utils/highlight.ts

```diff
@@ -15,23 +15,34 @@ export function createHighlight(rect: DOMRect, icon?: Icon) {
 	return highlight;
 }
 
-export function getHighlightZIndex(el: Element) {
+// Figures out the element's z-index and position, based on it's parents.
+export function getElementsPositionInDocument(el: Element) {
 	let highestZIndex = 0;
+	let fixed = false;
 	let current: Element | ParentNode | null = el;
 	while (current instanceof Element) {
-		let zIndex = Number(getComputedStyle(current).zIndex);
+		let style = getComputedStyle(current);
```

Comment:
I know this is not _worse_ than what we were doing before but **oof** I have performance concerns. 

IMO this approach is fine for now but I'd love to leave a comment that this is probably going to be one of the first places to look if we see performance issues.

Reply:
It only happens the first time. We probably should be wrapping the outer logic in a rAF in which case I think it's probably fine

Reply:
I'll add a comment

Reply:
To be clear though, I don't think there's any other way to know an element's z-index and position other than walking all the way up the DOM. If there's another way I'd definitely want to do it!